src/sampleutil.cpp

```diff
@@ -296,26 +296,31 @@ void SampleUtil::convertS16ToFloat32(CSAMPLE* _RESTRICT pDest, const SAMPLE* _RE
 }
 
 // static
-bool SampleUtil::sumAbsPerChannel(CSAMPLE* pfAbsL, CSAMPLE* pfAbsR,
+SampleUtil::CLIPPING SampleUtil::sumAbsPerChannel(CSAMPLE* pfAbsL, CSAMPLE* pfAbsR,
         const CSAMPLE* pBuffer, int iNumSamples) {
     CSAMPLE fAbsL = CSAMPLE_ZERO;
     CSAMPLE fAbsR = CSAMPLE_ZERO;
-    CSAMPLE clipped = 0;
+    CSAMPLE clippedL = 0;
+    CSAMPLE clippedR = 0;
```

Comment:
> Ah, you are right, that destroys vectorizing. I have just tested it with gcc 4.8

Can you briefly explain how to do this?



Can we just leave the current solution?
return static_cast<CLIP_STATUS>((clippedL>0) | ((clippedR>0)<<1));

An alternative would be:

    SampleUtil::CLIP_STATUS clipped = SampleUtil::NO_CLIPPING;
    if (clippedL > 0) {
        clipped = static_cast<SampleUtil::CLIP_STATUS>(clipped | SampleUtil::CLIPPING_LEFT);
    }
    if (clippedR > 0) {
        clipped = static_cast<SampleUtil::CLIP_STATUS>(clipped | SampleUtil::CLIPPING_RIGHT);
    }
    return clipped;

But I personally don't see an improvement on clarity. Whatever you prefer, though.